apps/api/src/app/widgets/usecases/remove-messages/remove-all-messages.usecase.ts

```diff
@@ -0,0 +1,137 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import {
+  MessageEntity,
+  DalException,
+  MessageRepository,
+  SubscriberRepository,
+  SubscriberEntity,
+  MemberRepository,
+  FeedRepository,
+} from '@novu/dal';
+import { ChannelTypeEnum } from '@novu/shared';
+import {
+  WsQueueService,
+  AnalyticsService,
+  InvalidateCacheService,
+  buildFeedKey,
+  buildMessageCountKey,
+} from '@novu/application-generic';
+
+import { RemoveAllMessagesCommand } from './remove-all-messages.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { MarkEnum } from '../mark-message-as/mark-message-as.command';
+
+@Injectable()
+export class RemoveAllMessages {
+  constructor(
+    private invalidateCache: InvalidateCacheService,
+    private messageRepository: MessageRepository,
+    private wsQueueService: WsQueueService,
+    private analyticsService: AnalyticsService,
+    private subscriberRepository: SubscriberRepository,
+    private memberRepository: MemberRepository,
+    private feedRepository: FeedRepository
+  ) {}
+
+  async execute(command: RemoveAllMessagesCommand): Promise<number> {
+    await this.invalidateCache.invalidateQuery({
+      key: buildFeedKey().invalidate({
+        subscriberId: command.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+    });
+
+    await this.invalidateCache.invalidateQuery({
+      key: buildMessageCountKey().invalidate({
+        subscriberId: command.subscriberId,
+        _environmentId: command.environmentId,
+      }),
+    });
+
+    const subscriber = await this.subscriberRepository.findBySubscriberId(command.environmentId, command.subscriberId);
+    if (!subscriber) throw new NotFoundException(`Subscriber ${command.subscriberId} not found`);
+
+    let deletedMessages;
+    let feed;
+    if (command.feedId !== '' && command.feedId !== null && command.feedId !== undefined) {
+      feed = await this.feedRepository.findById(command.feedId);
+      if (!feed) {
+        throw new NotFoundException(`Feed with ${command.feedId} not found`);
+      }
+    }
+    try {
+      const deleteMessageQuery: Partial<MessageEntity> = {
+        _environmentId: command.environmentId,
+        _organizationId: command.organizationId,
+        _subscriberId: subscriber._id,
+        channel: ChannelTypeEnum.IN_APP,
+      };
+
+      if (feed) {
+        deleteMessageQuery._feedId = feed._id;
+      }
```

Comment:
```suggestion
      const deleteMessageQuery: Partial<MessageEntity> = {
        _environmentId: command.environmentId,
        _organizationId: command.organizationId,
        _subscriberId: subscriber._id,
        channel: ChannelTypeEnum.IN_APP,
        _feedId: feed._id,
      };
```
should also work

Reply:
user can delete messages without feedId also

Reply:
`_feedId: feed?._id,` should work

Reply:
in that case _feedId will be undefined and incorrect values will be deleted

Reply:
from what I remember it will skip that `_feedId` in the query if it's undefined... but better to verify